### PR TITLE
detect/transform: from_base64 transform

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10991

This PR provides the `from_base64` transform used to match on content decoded with base64 using the mode (default rfc4648).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6487](https://redmine.openinfosecfoundation.org/issues/6487)

Describe changes:
- New transform: `from_base64`
- Documentation
- Keyword parser in rust.

Updates:
- Doc -- split extra long line and added ref to transform.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1670

### CODE REVIEW

**Review by jufajardini — APPROVED** [review]
Thanks for incorporating the doc changes! looking good to me :)

File: doc/userguide/rules/transforms.rst [hunk 1]
```diff
@@ -188,3 +188,53 @@ Example::
 
     alert http any any -> any any (msg:"HTTP ua only"; http.header_names; \
        bsize:16; content:"|0d 0a|User-Agent|0d 0a 0d 0a|"; nocase; sid:1;)
+
+.. _from_base64:
+
+from_base64
```
Comment (victorjulien) [inline doc/userguide/rules/transforms.rst:194]: can this explain what happens if data is not base64, or partly invalid? Will buffer be empty? Will invalid bytes be put in the buffer?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20460

File: rust/src/detect/transform_base64.rs [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+pub const DETECT_TRANSFORM_BASE64_FLAG_OFFSET_VAR: u8 = 0x08;
+pub const DETECT_TRANSFORM_BASE64_FLAG_NBYTES_VAR: u8 = 0x10;
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct SCDetectTransformFromBase64Data {
+    flags: u8,
+    nbytes: u32,
+    nbytes_str: *const c_char,
+    offset: u32,
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/detect/transform_base64.rs:85]: I think `u32` is correct as we can inspect larger than 64k buffers.

[the remark this answers is not in the document]
Comment (inashivb) [reply rust/src/detect/transform_base64.rs:85]: ok. Why is it an error then if the offest value exceeds 65535? https://github.com/OISF/suricata/pull/11007/files#diff-d8a96f1835ede0b3db0868c8e8d9f0f87cb75ac761a7a2232ac2b6d02a03ad97R187

**Review by catenacyber — CHANGES_REQUESTED** [review]
Now needs a rebase

File: rust/src/detect/transform_base64.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlucovsky) [reply rust/src/detect/transform_base64.rs:85]: If the offset can exceed 64k, then the `nbytes` value should too.

@victorjulien Is there a practical limit? u32::max seems excessive

[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/detect/transform_base64.rs:85]: I think this isn't consistently done throughout our keywords, I suggest checking some of the other keywords using an offset.

We should probably also create a ticket for refactoring / cleaning up these depth/offset like things.

[the remark this answers is not in the document]
Comment (jlucovsky) [reply rust/src/detect/transform_base64.rs:85]: [Task 7061](https://redmine.openinfosecfoundation.org/issues/7061) created for the clean up work.

For this keyword, I'm going to bound the value at 64k.

File: doc/userguide/rules/transforms.rst (hunk 1 above)
Comment (jlucovsky) [reply doc/userguide/rules/transforms.rst:194]: I'll add that to the description of the transform.

**Comment by jlucovsky** [comment]
Continued in #11299

**Pull request closed without merging**